### PR TITLE
AVX-56563 EAT interface update

### PR DESCRIPTION
**Pull request by sairane10**
- Updating the interfaces, adding EIP map, backup link info and ha interfaces for edge transit gateway
```
resource "aviatrix_transit_gateway" "edge-transit-5" {
    cloud_type = 262144
    account_name = "edge_aep"
    gw_name = "edge-transit-5"
    vpc_id = "site-25"
    gw_size = "SMALL"
    device_id = "763f27c3-20c6-46a4-9d2a-f408ee3c354a"
    local_as_number = "65457"
    bgp_polling_time = 33
    prepend_as_path = ["65457"]
    interfaces {
        gateway_ip    = "192.168.20.1"
        ip_address    = "192.168.20.11/24"
        type          = "WAN"
        index         = 0
        secondary_private_cidr_list = ["192.168.19.16/29"]
    }

    interfaces {
        gateway_ip    = "192.168.21.1"
        ip_address    = "192.168.21.11/24"
        type          = "WAN"
        index         = 1
        secondary_private_cidr_list = ["192.168.21.16/29"]
    }

    interfaces {
        dhcp   = true
        type   = "MANAGEMENT"
        index  = 0
    }

    interfaces {
        gateway_ip  = "192.168.22.1"
        ip_address  = "192.168.22.11/24"
        type        = "WAN"
        index       = 2
    }

    interfaces {
        gateway_ip = "192.168.23.1"
        ip_address = "192.168.23.11/24"
        type       = "WAN"
        index      = 3
    }

    interface_mapping{
      name  = "eth0"
      type  = "wan"
      index = 0
    }

    interface_mapping{
      name  = "eth1"
      type  = "wan"
      index = 1
    }

    interface_mapping{
      name  = "eth2"
      type  = "wan"
      index = 2
    }

    interface_mapping{
      name  = "eth3"
      type  = "mgmt"
      index = 0
    }

    interface_mapping{
      name  = "eth4"
      type  = "wan"
      index = 3
    }

    ha_device_id = "a20c75c0-06c2-4102-9df1-b00b85e89eac"
    peer_backup_port_type = "WAN"
    peer_backup_port_index = 1
    peer_connection_type = "private"
    ha_interfaces {
        gateway_ip    = "192.168.20.1"
        index         = 0
        ip_address    = "192.168.20.12/24"
        type          = "WAN"
    }

    ha_interfaces {
        gateway_ip   = "192.168.21.1"
        index        = 1
        ip_address   = "192.168.21.12/24"
        type         = "WAN"
        secondary_private_cidr_list = ["192.168.21.32/29"]
    }

    ha_interfaces {
        dhcp   = true
        index  = 0
        type   = "MANAGEMENT"
    }

    ha_interfaces {
        gateway_ip   = "192.168.22.1"
        index        = 2
        ip_address   = "192.168.22.12/24"
        type         = "WAN"
    }

    ha_interfaces {
        gateway_ip     = "192.168.23.1"
        index          = 3
        ip_address     = "192.168.23.12/24"
        type           = "WAN"
    }

    eip_map {
        interface_type = "WAN"
        interface_index = 0
        private_ip     = "192.168.19.18"
        public_ip      = "35.0.16.1"
    }
}
```

### CODE REVIEW

**Comment by stewartbutler** [comment]
This adds a whole lot of new functions with no tests. It seems that most of them are factored out of the original functions, but can we add some targeted tests for each of the new helpers?

**Comment by sairane10** [comment]
@stewartbutler I have added UT's for the new helper functions.

File: aviatrix/resource_aviatrix_transit_gateway.go [hunk 1]
```diff
@@ -3932,3 +4047,332 @@ func getInterfaceName(intfType string, intfIndex, wanCount int) (string, error)
...
+	}
+	interfaceListJson, err := json.Marshal(interfaceList)
+	if err != nil {
+		return "", err
+	}
+	interfacesEncoded := b64.StdEncoding.EncodeToString(interfaceListJson)
+	return interfacesEncoded, nil
+}
+
+func setInterfaceDetails(interfaces []goaviatrix.EdgeTransitInterface) []map[string]interface{} {
```
Comment (stewartbutler) [inline aviatrix/resource_aviatrix_transit_gateway.go:4321]: Need tests for this one too?

File: aviatrix/resource_aviatrix_transit_gateway.go [hunk 2]
```diff
@@ -3932,3 +4047,332 @@ func getInterfaceName(intfType string, intfIndex, wanCount int) (string, error)
...
+	eipMapJson, err := json.Marshal(eipMapStructured)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal eip_map to JSON: %w", err)
+	}
+
+	// Convert to a string with additional quotes for Terraform compatibility
+	return string(eipMapJson), nil
+}
+
+func getTransitHaGatewayDetails(d *schema.ResourceData, wanCount int, cloudType int) (*goaviatrix.TransitHaGateway, error) {
```
Comment (stewartbutler) [inline aviatrix/resource_aviatrix_transit_gateway.go:4122]: Does this need tests?

File: aviatrix/resource_aviatrix_transit_gateway.go [hunk 3]
```diff
@@ -2923,6 +2923,121 @@ func resourceAviatrixTransitGatewayUpdate(d *schema.ResourceData, meta interface
 
 	}
 
+	if goaviatrix.IsCloudType(gateway.CloudType, goaviatrix.EdgeRelatedCloudTypes) {
```
Comment (stewartbutler) [inline aviatrix/resource_aviatrix_transit_gateway.go:2836]: Can this be a separate function, just for readability? It seems it is all new content.

**Review by stewartbutler — CHANGES_REQUESTED** [review]
Two more functions need tests, unless you had a good reason to skip them.

Overall, I'd like to see more of the big `if` blocks decomposed into (preferably tested) functions -- this code is way too flat to follow. 

I'm concerned at how much refactor is going to be needed when we turn on golangci-strict next quarter -- a lot of these functions are badly violating `funlen` and `cyclop`, if nothing else, and it'll be necessary to fix that any time further changes are needed.

File: aviatrix/resource_aviatrix_transit_gateway.go (hunk 1 above)
Comment (sairane10) [reply aviatrix/resource_aviatrix_transit_gateway.go:4321]: Added the UT for this.

File: aviatrix/resource_aviatrix_transit_gateway.go (hunk 2 above)
Comment (sairane10) [reply aviatrix/resource_aviatrix_transit_gateway.go:4122]: This function contains the helper function for which I have already added UT's.For the backuplink info details, I have updated the AEP config to verify that these values are being set as expected.

File: aviatrix/resource_aviatrix_transit_gateway.go (hunk 3 above)
Comment (sairane10) [reply aviatrix/resource_aviatrix_transit_gateway.go:2836]: I have added a separate createEdgeTransit function for this if block which creates primary gateway, ha gateway, updates eip_map for edge transit and sets the bgp values.

File: docs/resources/aviatrix_transit_gateway.md [hunk 4]
```diff
@@ -320,60 +323,81 @@ resource "aviatrix_transit_gateway" "test-edge-transit-1" {
...
+      index = 0
     }
-    interfaces {
-        gateway_ip                  = "192.168.21.1"
-        name                        = "eth1"
-        ip_address                  = "192.168.21.12/24"
-        type                        = "WAN"
+    interface_mapping{
+      name  = "eth1"
+      type  = "wan"
```
Comment (vemurisaikartikeya) [inline docs/resources/aviatrix_transit_gateway.md]: The type of the interface is different for interface config and interface mapping.
I know it is the same like this in the API as well.
But do you want to make it common at both places in TF in upper case like:
"WAN". "LAN", "MANAGEMENT"

File: aviatrix/resource_aviatrix_transit_gateway_test.go [hunk 5]
```diff
@@ -362,39 +445,81 @@ resource "aviatrix_transit_gateway" "test_transit_gateway_aep" {
...
-        type                        = "WAN"
-        secondary_private_cidr_list = ["192.168.18.112/29"]
+        gateway_ip = "192.168.23.1"
+        ip_address = "192.168.23.11/24"
+        type       = "WAN"
+        index      = 3
+    }
+
+	ha_device_id = "a20c75c0-06c2-4102-9df1-b00b85e89eac"
+    peer_backup_port_type = "WAN"
```
Comment (vemurisaikartikeya) [inline aviatrix/resource_aviatrix_transit_gateway_test.go:487]: I think this is perfectly fine.
But based on some of the discussions today, people were suggesting to use an attibute combining both type and index like:
LAN0, WAN1 etc.
I understand that is like this for interface config, specified separately.
But for other pieces of config, just get Shiva's opinion as well or if that can be done later, then having this currently is fine as well.

**Review by vemurisaikartikeya — APPROVED** [review]

File: docs/resources/aviatrix_transit_gateway.md (hunk 4 above)
Comment (sairane10) [reply docs/resources/aviatrix_transit_gateway.md]: Updated the interface mapping to match interface config type name

File: aviatrix/resource_aviatrix_transit_gateway_test.go (hunk 5 above)
Comment (sairane10) [reply aviatrix/resource_aviatrix_transit_gateway_test.go:487]: I think that will require some significant code change in the getter and setter function. Can include that change based on what Shiva has to say.

**Review by stewartbutler — APPROVED** [review]

**Pull request merged**